### PR TITLE
 add insights metrics deployment

### PR DESCRIPTION
**Pull request by xrajesh**
/hold
`/hold`

### CODE REVIEW

**Review by nathanweatherly — APPROVED** [review]

File: stable/insights-chart/templates/insights-metrics-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2021 Red Hat, Inc.
+# Copyright Contributors to the Open Cluster Management project
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "insights.fullname" . }}-insights-metrics
```
Comment (jlpadilla) [inline stable/insights-chart/templates/insights-metrics-deployment.yaml:7]: I find that static names are much easier to use in automation.  There's the concern of having name collision, but we haven't hit this problem with the search deployments with static names (search-operator). 
```suggestion
  name: ocm-insights-metrics
```

File: stable/insights-chart/templates/insights-metrics-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            - ALL
+        args:
+        - --port=8080
+        - --telemetry-port=8081
+        resources:
+{{ toYaml .Values.insights.metrics.resources | indent 10 }}
+        livenessProbe:
+          exec:
+            command:
+            - ls
```
Comment (jlpadilla) [inline stable/insights-chart/templates/insights-metrics-deployment.yaml:76]: Is there an API endpoint we could use instead?

File: stable/insights-chart/values.yaml [hunk 3]
```diff
@@ -54,3 +77,10 @@ insights:
         cpu: "25m"
       limits:
         memory: "512Mi"
+  metrics:
+    resources:
+      requests:
+        memory: "32Mi"
+        cpu: "25m"
+      limits:
+        memory: "512Mi"
```
Comment (jlpadilla) [inline stable/insights-chart/values.yaml:86]: The limit seems high compared to the request, maybe it's ok, but just checking.

File: stable/insights-chart/templates/insights-metrics-deployment.yaml (hunk 2 above)
Comment (xrajesh) [reply stable/insights-chart/templates/insights-metrics-deployment.yaml:76]: Yes , I am enabling TLS .. next PR @jlpadilla

File: stable/insights-chart/values.yaml (hunk 3 above)
Comment (xrajesh) [reply stable/insights-chart/values.yaml:86]: Yes , I have to get a sweet spot

File: stable/insights-chart/templates/insights-metrics-deployment.yaml (hunk 1 above)
Comment (xrajesh) [reply stable/insights-chart/templates/insights-metrics-deployment.yaml:7]: Thank you - Will Update it

**Pull request merged**